src/libcollections/string.rs

```diff
@@ -809,14 +986,23 @@ impl String {
         self.len() == 0
     }
 
-    /// Truncates the string, returning it to 0 length.
+    /// Truncates this `String`, removing all contents.
+    ///
+    /// While this means the `String` will have a length of zero, it does not
+    /// touch its capacity.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
-    /// let mut s = "foo".to_string();
+    /// let mut s = String::from("foo");
```

Comment:
Right now, they're split across a lot of ways. We really need to decide what the actual convention is, there's a lot of documentation that will end up changing no matter what is chosen.